programs/drift/src/controller/orders.rs

```diff
@@ -1128,58 +1130,76 @@ pub fn fill_perp_order(
     Ok((base_asset_amount, updated_user_state))
 }
 
-pub fn validate_market_within_price_band(
+pub fn validate_perp_market_within_price_band(
     market: &PerpMarket,
     state: &State,
     potentially_risk_increasing: bool,
-    oracle_reserve_price_spread_pct_before: Option<i64>,
+    oracle_ref_price_spread_pct_before: Option<i64>,
+    ref_price_after: Option<u64>,
 ) -> DriftResult<bool> {
-    let reserve_price_after = market.amm.reserve_price()?;
-
-    let is_oracle_mark_too_divergent_before = if let Some(oracle_reserve_price_spread_pct_before) =
-        oracle_reserve_price_spread_pct_before
-    {
-        amm::is_oracle_mark_too_divergent(
-            oracle_reserve_price_spread_pct_before,
-            &state.oracle_guard_rails.price_divergence,
-        )?
+    let ref_price_after = if let Some(ref_price_after) = ref_price_after {
+        ref_price_after
     } else {
-        false
+        market.amm.reserve_price()?
     };
 
-    let oracle_reserve_price_spread_pct_after =
-        amm::calculate_oracle_twap_5min_mark_spread_pct(&market.amm, Some(reserve_price_after))?;
+    let default_oracle_guard_rail_divergence: u64 = state
+        .oracle_guard_rails
```

Comment:
might be nice to add fn to oracle guardrails that does this mul/div for you to make it more readable

Reply:
done